Tasks/AzureRmWebAppDeployment/kuduutility.ts

```diff
@@ -133,4 +133,45 @@ export async  function containsParamFile(webAppPackage: string ) {
     }
     tl.debug(tl.loc("Isparameterfilepresentinwebpackage0", isParamFilePresent));
     return isParamFilePresent;
+}
+
+export async function createKuduPhysicalPath(publishingProfile, physicalPath: string) {
+    var defer = Q.defer<string>();
+    physicalPath = physicalPath.replace(/[\\]/g, "/");
+    var kuduPhysicalpathUrl = "https://" + publishingProfile.publishUrl + "/api/vfs/" + physicalPath + "/";
+    var basicAuthToken = 'Basic ' + new Buffer(publishingProfile.userName + ':' + publishingProfile.userPWD).toString('base64');
+    var headers = {
+        'Authorization': basicAuthToken,
+        'If-Match': "*"
+    };
+    tl.debug(tl.loc("RequestedURLforkuduphysicalpath", kuduPhysicalpathUrl))
+
+    httpObj.send('GET', kuduPhysicalpathUrl, null, headers, (error, response, body) => {
+        if (error) {
+            defer.reject(error);
+        }
```

Comment:
I believe you are first checking and then creating, Divide them into two functions to make function smaller.